### ISSUE TITLE
Unable to remove stale report

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#50061]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @flodnv 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1727864292272169

## Action Performed:
Break down in numbered steps

Create a report and ensure it fails - this _could_ be following
1. Have a workspace with 2 members and a policy room
2. As user A post a message in the room
3. As User A, go offline
4. As User A, while offline, create a thread from that comment and post a comment there
5. As user B, online, create a thread from the above comment
6. Post a comment in that thread
7. As user A, go online

## Expected Result:
Describe what you think should've happened

For user A, this stale report should have a GBR if it failed to create using the optimistic IDs. Clicking into the report, you should be able to see an explanatory error message and X button that will clear the chat

## Actual Result:
Describe what actually happened

The GBR is showing in LHN, but there is no explanation or option to clear the chat from onyx

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by rinej** [comment]
It's Tomek for callstack, I will investigate this issue

**Comment by mountiny** [comment]
@rinej what are your findings?

**Comment by rinej** [comment]
I reviewed the `handleUnusedOptimisticID` function, and in this case, we don’t follow that flow because the server does not return a `preexistingReportID` in the report. In our situation, when reopening a report after returning from offline mode (with the thread already created), we receive a response with an error:
```
{"code":666,"jsonCode":666,"type":"Expensify\\Libs\\Error\\ExpError","UUID":"E490CD6D-46BF-4EE2-8B4F-51672557284A","message":"You don't have access to this chat room.","title":"Room not accessible","data":{"onyxData":[{"onyxMethod":"merge","key":"reportActions_1419398802104800","value":{"290042788771067714":{"errors":{"1727893908883670":"You don't have access to this chat room."}}}}]},"htmlMessage":"","onyxData":[{"onyxMethod":"merge","key":"reportActions_1419398802104800","value":{"290042788771067714":{"errors":{"1727893908883670":"You don't have access to this chat room."}}}}],"requestID":"8cc6c5816c3bb1f7-WAW"}
So we treat it just as another report with the error. In the app, we create a copy of the report without removing the original, so the old report still appears in the LHN. Even though the "X" clears the error label, the report remains in the list.
```

So we treat it just as another report with the error. In the app, we create a copy of the report without removing the original, so the old report still appears in the LHN. Even though the "X" clears the error label, the report remains in the list.

After logging out and logging in again the stale reports are properly removed from onyx state.

Here are the next steps we can check

- It could be an issue with the backend, where the `preexistingReportID` field should be set in our scenario, triggering the `handleUnusedOptimisticID` flow.
- On our end, we can further investigate how to remove the report with the error after creating the copy, probably add some check in `shouldReportBeInOptionList`

I am investigating the second steps currently

**Comment by rinej** [comment]
@mountiny 
after recent changes on main I am not able to reproduce the issue. 
When following the qa steps, I am able to remove the stale report with the X button, and the proper one (the message which is send from online) is taking place. It looks like it works properly. I am adding the video and screenshot with example. Is it the expected behaviour? 

https://github.com/user-attachments/assets/74fffbc9-8261-48d2-ad22-7cdcbff77588


<img width="996" alt="Screenshot 2024-10-08 at 13 30 30" src="https://github.com/user-attachments/assets/42674bb3-f3da-4cef-a8a3-590885c8110f">

**Comment by mountiny** [comment]
Yeah, I think this is good; it might be some regression around the RBR handling, but it seems like we cleared it.

I will close this issue for now and we can reopen if this is a problem again
